app/controllers/discussions_controller.rb

```diff
@@ -48,14 +51,25 @@ def update
 
   private
 
+    def check_discussion_is_upcoming
+      if !(@discussion.date.to_s > Date.current.to_s)
```

Comment:
Also, you can make a method in `discussion` model. `upcoming?` and use it wherever required.